animate/interpolation.py

```diff
@@ -326,15 +326,15 @@ def clement_interpolant(source, target_space=None, boundary=False):
     # Process target space
     Vt = target_space
     if Vt is None:
-        if rank == 0:
-            Vt = firedrake.FunctionSpace(mesh, "CG", 1)
-        elif rank == 1:
-            Vt = firedrake.VectorFunctionSpace(mesh, "CG", 1)
-        else:
-            Vt = firedrake.TensorFunctionSpace(mesh, "CG", 1)
+        Vt_ranks = {
+            0: firedrake.FunctionSpace(mesh, "CG", 1),
+            1: firedrake.VectorFunctionSpace(mesh, "CG", 1),
+            2: firedrake.TensorFunctionSpace(mesh, "CG", 1),
+        }
+        Vt = Vt_ranks[rank]
```

Comment:
This is less efficient because now we construct 3 spaces rather than 1. How about this?
```suggestion
    Vt = {
        0: firedrake.FunctionSpace,
        1: firedrake.VectorFunctionSpace,
        2: firedrake.TensorFunctionSpace,
    }[rank](mesh, "CG", 1)
    ```

Reply:
Thanks! Done as you suggested in 2946fc8a3241de5d1fa4732cda88d6a1ad16c0cf